### PR TITLE
UP-3731 - Implements first storybook test

### PR DESCRIPTION
**Pull request by gustavoaz7**

### CODE REVIEW

File: packages/react-slider/src/components/slider.stories.tsx [hunk 1]
```diff
@@ -29,7 +31,60 @@ const meta: Meta<typeof RecommendationsSlider> = {
...
+    await waitFor(function recosLoaded() {
+      expect(sliderEl.querySelector('tfup-recommendation')).toBeTruthy();
+    }, {timeout: 2000});
+
+    const [ arrowRight, arrowLeft ] = [
+      sliderEl.querySelector<HTMLButtonElement>('[aria-label="arrow-right"]'),
+      sliderEl.querySelector<HTMLButtonElement>('[aria-label="arrow-left"]'),
+    ];
+    const [ firstReco, secondReco, thirdReco ] = Array
+      .from(sliderEl.querySelectorAll('tfup-recommendation'));
```
Comment (enjoythelive1) [inline packages/react-slider/src/components/slider.stories.tsx:56]: This should not be needed for destructuring. NodeList implements iteration and any iterable can be deconstructed.
```suggestion
    const [
      firstReco,
      secondReco,
      thirdReco,
    ] = sliderEl.querySelectorAll('tfup-recommendation');
```

File: packages/slider/src/components/recommendations-slider/recommendations-slider.ts [hunk 2]
```diff
@@ -272,7 +272,7 @@ export class RecommendationsSlider extends LitElement {
         <button
           class="arrow"
           @click=${this.#scrollToPrevious}
-          aria-hidden="true"
+          aria-label="arrow-left"
```
Comment (enjoythelive1) [inline packages/slider/src/components/recommendations-slider/recommendations-slider.ts:275]: Besides a way to select the buttons, why?

**Review by enjoythelive1 — CHANGES_REQUESTED** [review]
Why were these tests made for the react one instead of the actual component?

**Comment by gustavoaz7** [comment]
> Why were these tests made for the react one instead of the actual component?

This is a PoC and I opted for what will most likely be used the most

File: packages/slider/src/components/recommendations-slider/recommendations-slider.ts (hunk 2 above)
Comment (gustavoaz7) [reply packages/slider/src/components/recommendations-slider/recommendations-slider.ts:275]: Accessibility

File: packages/react-slider/src/components/slider.stories.tsx (hunk 1 above)
Comment (gustavoaz7) [reply packages/react-slider/src/components/slider.stories.tsx:56]: > Type 'NodeListOf<Recommendation>' is not an array type.ts(2461)

Comment (enjoythelive1) [reply packages/react-slider/src/components/slider.stories.tsx:56]: This must be because of the compilation target. I guess its ok

**Review by enjoythelive1 — APPROVED** [review]

File: packages/react-slider/src/components/slider.stories.tsx [hunk 3]
```diff
@@ -29,7 +31,60 @@ const meta: Meta<typeof RecommendationsSlider> = {
...
+    await step('Slide right once', async () => {
+      arrowRight.click();
+
+      await waitFor(function firstRecoOut() {
+        expect(isRecoVisible(firstReco)).toBeFalsy();
+        expect(isRecoVisible(secondReco)).toBeTruthy();
+      }, {timeout: 2000});
+    });
+
+    await step('Slide right twice', async () => {
```
Comment (Drafter500) [inline packages/react-slider/src/components/slider.stories.tsx:67]: so each next test depends on the previous one?

**Review by Drafter500 — APPROVED** [review]
The code looks fine to me, but I could not tell is its useful?
@gustavoaz7 What are your thoughts on it? Did you like it, the API, and the structure? Do you see real benefits? 

Btw, When running locally I get this error `Cannot read properties of null (reading 'click')` and the test fails, do you have a clue?

**Comment by enjoythelive1** [comment]
> The code looks fine to me, but I could not tell is its useful? @gustavoaz7 What are your thoughts on it? Did you like it, the API, and the structure? Do you see real benefits?
> 
> Btw, When running locally I get this error `Cannot read properties of null (reading 'click')` and the test fails, do you have a clue?

I see this is useffull as it could replace the test we have in the slider about testing this specific behavior. The cypress test setup is hacky and complicated. It would be easier to run these tests in the command line.

File: packages/react-slider/src/components/slider.stories.tsx (hunk 3 above)
Comment (gustavoaz7) [reply packages/react-slider/src/components/slider.stories.tsx:67]: "test" is the whole `play` function. `step`s are just used to group interactions (https://storybook.js.org/docs/react/writing-tests/interaction-testing#group-interactions-with-the-step-function). It also makes the test output cleaner in storybook

**Comment by enjoythelive1** [comment]
Also, in web compoenents. It would be a better test platform for web components. As we could see the behavior in storybook and also have the tests run in a real browser environmet. But that would also make tests more expensive.

We could use it as part of what we show in stories, as a way to showcase behavior of components (like dropdown opening) or an input being used.

**Review by uschischroder — APPROVED** [review]

**Review by Drafter500 — APPROVED** [review]

**Pull request merged**